sqlglot/generator.py

```diff
@@ -4268,6 +4270,14 @@ def _simplify_unless_literal(self, expression: E) -> E:
         return expression
 
     def _embed_ignore_nulls(self, expression: exp.IgnoreNulls | exp.RespectNulls, text: str) -> str:
+        this = expression.this
+        if isinstance(this, self.RESPECT_IGNORE_NULLS_UNSUPPORTED_EXPRESSIONS):
+            this_name = this.sql_name() if isinstance(this, exp.Func) else this.__class__
```

Comment:
Nit: relying on `sql_name()` may lead to printing a function name that is not found in the input, since we sometimes abstract away and use SQLGlot-specific expression types.

Should we instead print `type(expression).key`? Or `split` on `"("` and use the first segment as the function's name.